sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -511,6 +511,9 @@ def test_precompute_invalid_argument():
                 LassoCV(precompute="invalid")]:
         assert_raises(ValueError, clf.fit, X, y)
 
+    # Precompute = 'auto' is not supported for ElasticNet
+    assert_raises(ValueError, ElasticNet(precompute='auto').fit, X, y)
```

Comment:
assert_raises_regex should always be preferred I think because you can check the error message as well as the type.

I have bumped into cases where the type of the error raised was right only by chance ...